### PR TITLE
post push karma policy checking

### PR DESCRIPTION
**Pull request by lmacken**
...and some other stuff.

### CODE REVIEW

File: bodhi/models/models.py [hunk 1]
```diff
@@ -1553,6 +1522,24 @@ def check_requirements(self, session, settings):
...
+                    self.request = UpdateRequest.stable
+                    self.date_pushed = None
+                elif self.unstable_karma != 0 and self.karma <= self.unstable_karma:
+                    log.info("Automatically unpushing %s" % self.title)
+                    self.obsolete()
+            else:
+                # Ignore karma thresholds for pending/stable/obsolete updates
+                pass
+        else:
+            log.debug('%s locked. Ignoring karma thresholds.' % self.title)
```
Comment (ralphbean) [inline bodhi/models/models.py]: Seems like this should raise a LockedUpdateException.

File: bodhi/tests/test_masher.py [hunk 2]
```diff
@@ -440,13 +439,22 @@ def test_security_update_priority(self, publish, *args):
 
         # Ensure that F18 runs before F17
         calls = publish.mock_calls
+        # Order of fedmsgs at the the moment:
```
Comment (ralphbean) [inline bodhi/tests/test_masher.py]: Nice comment.

File: bodhi/tests/test_masher.py [hunk 3]
```diff
@@ -843,3 +851,62 @@ def test_resume_push(self,  *args):
...
+    @mock.patch('bodhi.util.cmd')
+    def test_stable_requirements_met_during_push(self,  *args):
+        """
+        Test reaching the stablekarma threshold while the update is being
+        pushed to testing
+        """
+        title = self.msg['body']['msg']['updates']
+
+        # Simulate a failed push
+        with mock.patch.object(MasherThread, 'verify_updates', mock_exc):
```
Comment (ralphbean) [inline bodhi/tests/test_masher.py]: Very cool.

**Comment by ralphbean** [comment]
:+1: barring that ``LockedUpdateException`` comment.

Looks good!

**Comment by lmacken** [comment]
Thanks!!

**Pull request merged**